src/test/java/com/systelab/seed/patient/PatientResourceTest.java

```diff
@@ -279,47 +280,32 @@ public void testDeleteUnexistingPatient() {
     @Test
     public void testUpdatePatient()
     {
-        Patient patient = getPatientData("John", "Burrows", "jburrows@werfen.com");
-        Patient patientCreated = given().body(patient)
-            .when().post("/patients/patient")
-            .then().assertThat().statusCode(200)
+        Patient patient = getPatientData("John", "Burrows", "jburrows@werfen.com", null,null, null,null,null);
+        Response response = doCreatePatient(patient);
+        Patient patientCreated = response.then().assertThat().statusCode(200)
             .extract().as(Patient.class);
         Assertions.assertNotNull(patientCreated, "Patient not created");
         patientCreated.setEmail("new@emailchanged.com");
-        Patient patientUpdated = given().body(patientCreated)
-            .when().put("/patients/" + patientCreated.getId())
-            .then().assertThat().statusCode(200)
+        UUID patientCreatedId = patientCreated.getId();
+
+        Response responseUpdated = doUpdatePatient(patient, patientCreatedId);
+        Patient finalCheck = responseUpdated.then().assertThat().statusCode(200)
             .extract().as(Patient.class);
-        Assertions.assertNotNull(patientUpdated, "Patient not updated");
-
-        TestUtil.checkField("Email", patientCreated.getEmail(), patientUpdated.getEmail());
-        TestUtil.checkField("Name", patientCreated.getName(), patientUpdated.getName());
-        TestUtil.checkField("Surname", patientCreated.getSurname(), patientUpdated.getSurname());
-        TestUtil.checkField("Medical Number", patientCreated.getMedicalNumber(), patientUpdated.getMedicalNumber());
-        TestUtil.checkField("DoB", patientCreated.getDob(), patientUpdated.getDob());
-        TestUtil.checkField("Street", patientCreated.getAddress().getStreet(), patientUpdated.getAddress().getStreet());
-        TestUtil.checkField("City", patientCreated.getAddress().getCity(), patientUpdated.getAddress().getCity());
-        TestUtil.checkField("Zip", patientCreated.getAddress().getZip(), patientUpdated.getAddress().getZip());
+        Assertions.assertNotNull(responseUpdated, "Patient not updated");
+        checkPatientData(patientCreated, finalCheck);
```

Comment:
This test is failing: you are compairing the original patient with updated patient. If the update has been done correctly this is never going to be coincident.